crates/paimon/src/spec/types.rs

```diff
@@ -243,12 +294,54 @@ impl BigIntType {
     }
 }
 
+impl Serialize for BigIntType {
+    fn serialize<S>(
+        &self,
+        serializer: S,
+    ) -> Result<<S as serde::Serializer>::Ok, <S as serde::Serializer>::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
+impl<'de> Deserialize<'de> for BigIntType {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_str(BigIntTypeVisitor)
```

Comment:
I prefer to implement `from_str` instead and forward `Deserialize` to `from_str`.